File: examples/surface-protocol/openai.py
from runx import RunxClient, create_openai_surface_adapter, create_surface_bridge


def main() -> None:
    client = RunxClient()
    bridge = create_surface_bridge(client)
    adapter = create_openai_surface_adapter(bridge)
    response = adapter.run(
        "skills/sourcey",
        inputs={"project": "."},
        resolver=lambda context: True if context.request.get("kind") == "approval" else None,
    )
    print(response)


if __name__ == "__main__":
    main()

File: examples/surface-protocol/openai.ts
import { createRunxSdk, createSurfaceBridge } from "@runx/sdk";
import { createOpenAiSurfaceAdapter } from "@runxhq/host-adapters";

async function main(): Promise<void> {
  const sdk = createRunxSdk({ callerOptions: { maxAttempts: 1 } });
  const bridge = createSurfaceBridge({ execute: sdk.runSkill.bind(sdk) });
  const openai = createOpenAiSurfaceAdapter(bridge);

  const response = await openai.run({
    skillPath: "skills/sourcey",
    inputs: { project: "." },
    resolver: ({ request }) => {
      if (request.kind === "approval") {
        return true;
      }
      return undefined;
    },
  });

  console.log(JSON.stringify(response, null, 2));
}

void main();
